engine/classes/Elgg/Gatekeeper.php

```diff
@@ -191,6 +192,19 @@ public function assertAccessibleEntity(ElggEntity $entity, ElggUser $user = null
 				throw $exception;
 			}
 
+			$user_guid = isset($user) ? (int) $user->guid : _elgg_services()->session->getLoggedInUserGuid();
```

Comment:
replace `_elgg_services()->session->getLoggedInUserGuid()` with `$this->session->getLoggedInUserGuid()` please